### PR TITLE
Update docs to more throughly describe the backend AWS deployment process

### PR DESCRIPTION
**Pull request by hankewyczz**
@sebwittr

### CODE REVIEW

File: docs/deployment/deploy.md [hunk 1]
```diff
@@ -3,20 +3,47 @@
...
+### Steps
 
 - Run `./infrastructure/aws/push-image`
-  - This pushes a new Docker image to AWS ECR with the `staging` tag
+
+This runs `docker build` in your local environment, tags the resulting image, and pushes it up to AWS ECR (a registry for Docker images). This also tags the Docker image with the `staging` tag.
+
+We maintain two clusters (ie. groups) of ECS (Elastic Container Service) containers running in AWS — a staging group, and a production group. Whenever the staging group is restarted, it will use the Docker image with the `staging` tag as its source. However, instead of waiting for the staging group to redeploy itself (which would only happen when the container runs into an error and dies), we forcibly restart it in the next step.
+
+?> Occasionally, you may run into deployment issues where the image deployed to AWS doesn't accuratly reflect the state it should. The cause is currently unknown, but it's suspected to be a Docker caching issue. Try using `docker prune` to remove your current images and caches.<br/><br/>2023-02-13: @sebwittr is investigating.
```
Comment (Lucas-Dunker) [inline docs/deployment/deploy.md:34]: What is the use of ?> here? Also, is the 2023-02-13 investigation resolved?

Comment (hankewyczz) [reply docs/deployment/deploy.md:34]: `?>` renders as a tip box — check out the deployed preview.

For the second part, @sebwit20

Comment (Lucas-Dunker) [reply docs/deployment/deploy.md:34]: image


This is what it's showing in the preview, is this correct?

Comment (hankewyczz) [reply docs/deployment/deploy.md:34]: It's not a Github feature, it's a [Docsify feature](https://docsify.js.org/#/helpers?id=general-tips) (the renderer we use for our documentation websites)

**Review by Lucas-Dunker — APPROVED** [review]
Looks good!

**Review by pranavphadke1 — APPROVED** [review]

**Pull request merged**